src/lightning/fabric/utilities/spike.py

```diff
@@ -0,0 +1,167 @@
+import json
+import operator
+import os
+import warnings
+from typing import Any, Dict, List, Literal, Optional, TYPE_CHECKING, Union
+
+import torch
+from lightning_utilities.core.imports import compare_version
+
+_TORCHMETRICS_GREATER_EQUAL_1_0_0 = compare_version("torchmetrics", operator.ge, "1.0.0")
+
+if TYPE_CHECKING:
+    from lightning.fabric.fabric import Fabric
+
+from lightning.fabric.utilities.types import _PATH
+
+
+class SpikeDetection:
+    """Spike Detection Callback.
+
+    Terminates training with a ``TrainingSpikeException`` when a loss-spike was detected and
+    saves the batches to skip when resuming to a file.
+
+    We skip the current and the previous batch since it is unclear whether the previous batch
```

Comment:
In the future we should make this configurable so the user can decide how many batches to skip. I'm expecting this to be application-dependent.